### PR TITLE
Allow bundling in process instead of using a worker

### PR DESCRIPTION
**Pull request by theneva**
## JIRA Issue
N/A

## Status
**READY**

## Description
Allow bundling in process instead of using a worker process. This is helpful when you don't want to clean up after the server is no longer used, particularly when testing a Podlet or Layout server with a local instance of the asset server.

I have:

- Pulled the bundling stuff from `lib/utils` out into a class (`lib/bundler.js`)
- Extended that class with the optimistic bundker
- Added tests to verify that ending workers functions as intended regardless of whether the worker is used (in a very `java.util.*` way)
- Made `endWorkers` async

## Todos
- [x] Tests
- [ ] Documentation (not entirely sure if needed)

## Deploy Notes
N/A

## Related PRs
* None

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -9,128 +9,136 @@ const parseJson = require('parse-json');
...
-        return await Promise.all(
-            ids.map(async fileName => ({
-                fileName,
-                contents: await sink.get(fileName),
-            }))
-        );
-    } catch (err) {
-        throw Boom.boomify(err, {
-            message: 'Unable to fetch 1 or more feeds.',
+    if (bundleInProcess) {
```
Comment (SimenB) [inline lib/utils.js]: I think we should just move bundling out of `utils`.

File: lib/utils.js [hunk 2]
```diff
@@ -9,128 +9,136 @@ const parseJson = require('parse-json');
...
+        if (value === 'true' || value === true) return true;
+        if (value === 'false' || value === false) return false;
+        return defaultValue;
+    }
 
-function endWorkers() {
-    bundleWorker.end();
-}
+    function endWorkers() {
+        bundler.end();
```
Comment (SimenB) [inline lib/utils.js]: `end` is only available when it's a worker, need to check here

File: lib/utils.js (hunk 1 above)
Comment (theneva) [reply lib/utils.js]: works for me

File: lib/utils.js (hunk 2 above)
Comment (theneva) [reply lib/utils.js]: true

File: lib/bundler.js [hunk 3]
```diff
@@ -1,7 +1,133 @@
...
+        return result.value;
+    }
+
+    endWorkers() {
+        if (this.bundleInProcess) {
+            // TODO: Notify the user that this operation is pointless?
+            return;
+        }
+
+        this.bundler.end();
```
Comment (SimenB) [inline lib/bundler.js]: return it? I know it's `void`, but it might have a promise later

Comment (SimenB) [reply lib/bundler.js]: or make the function `async` and `await`

**Pull request merged**